gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -68,6 +69,7 @@
 
     private final CodeableConceptMapper codeableConceptMapper;
     private final ConsultationListMapper consultationListMapper;
+    private final ImmunizationChecker immunizationChecker;
```

Comment:
I would change the ResourceReferenceUtil methods so they're no longer static and inject ImmunizationChecker there, not here.